apps/test-app/frontend/src/components/app/App.tsx

```diff
@@ -77,13 +81,28 @@ export function App() {
       }
 
       // determine what the viewport is hiliting
-      const hiliteSet = await Presentation.selection.getHiliteSet(args.imodel);
-      if (hiliteSet.elements) {
-        // note: the hilite list may contain models and subcategories as well - we don't
-        // care about them at this moment
-        await IModelApp.viewManager.selectedView.zoomToElements(hiliteSet.elements);
-      }
+      const selectedView = IModelApp.viewManager.selectedView;
+      const elementIds: Id64String[] = [];
+      from(Presentation.selection.getHiliteSetIterator(args.imodel))
+        .pipe(takeUntil(cancel))
+        .subscribe({
+          next: (set) => {
+            if (!set.elements) {
+              // note: the hilite list may contain models and subcategories as well - we don't
+              // care about them at this moment
+              return;
+            }
+
+            elementIds.push(...set.elements);
+            void selectedView.zoomToElements(elementIds);
```

Comment:
I'm trying to imagine how zooming to elements would look like when we load them in batches. And don't think that's a good idea. Have you tried that?